.github/workflows/pull-request.yml

```diff
@@ -272,6 +272,132 @@ jobs:
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '22.x'
+
+      - name: Cache Node.js dependencies
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.npm
+            node_modules
+          key: ${{ runner.os }}-docker-check-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-docker-check-
+
+      - name: Install Docker Compose
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y docker-compose
+
+      - name: Check if Talawa API starts in Docker
+        run: |
+          # Ensure no containers are running
+          docker-compose -f docker-compose.dev.yaml down -v || true
+
+          # Verify docker-compose file exists
+          if [ ! -f "docker-compose.dev.yaml" ]; then
+            echo "Error: docker-compose.dev.yaml not found"
+            exit 1
+          fi
+
+          # Start containers
+          if ! timeout 300 docker-compose -f docker-compose.dev.yaml up -d --build; then
+            echo "Failed to start containers"
+            docker-compose -f docker-compose.dev.yaml logs
+            exit 1
+          fi
+
+          # Wait for MongoDB and Redis to be ready
+          echo "Waiting for MongoDB..."
+          timeout=30
+          until docker-compose -f docker-compose.dev.yaml exec -T mongo mongosh --eval "db.runCommand({ ping: 1 }).ok">/dev/null 2>&1 || [ $timeout -eq 0 ]; do
+            echo "Waiting for MongoDB to be ready..."
+            sleep 1
+            ((timeout--))
+          done
+          if [ $timeout -eq 0 ]; then
+            echo "Error: MongoDB failed to start within timeout"
+            echo "Fetching MongoDB logs..."
+            docker-compose -f docker-compose.dev.yaml logs mongodb
+            echo "Shutting down MongoDB..."
+            docker-compose -f docker-compose.dev.yaml down -v
+            exit 1
+          else
+            echo "MongoDB is ready!"
+          fi
+
+          echo "Waiting for Redis..."
+          timeout=30
+          until docker-compose -f docker-compose.dev.yaml exec -T redis-stack-server redis-cli ping >/dev/null 2>&1 || [ $timeout -eq 0 ]; do
+            sleep 1
+            ((timeout--))
+          done
+          if [ $timeout -eq 0 ]; then
+            echo "Error: Redis failed to start within timeout"
+            echo "Fetching Redis logs..."
+            docker-compose -f docker-compose.dev.yaml logs redis-stack-server
+            echo "Shutting down Redis..."
+            docker-compose -f docker-compose.dev.yaml down -v
+            exit 1    
+          else
+            echo "Redis is ready!"
+          fi
+
+          # Wait for TALAWA API to be healthy 
+          timeout=60
+          until docker-compose -f docker-compose.dev.yaml exec -T talawa-api-dev curl -v -X OPTIONS "http://talawa-api-dev:4000/graphql" 2>&1 || [ $timeout -eq 0 ]; do
```

Comment:
Please test the api with a query which returns a specific response. Since in the issue I was working initially the `/graphql` was working but was not accepting any queries.